### PR TITLE
Change dumps to AnyStr

### PR DESCRIPTION
**Pull request by ahopkins**
A dumps method that returns a bytes string (like orjson.dumps) is valid. So, the type annotation needs to be `AnyStr`.

### CODE REVIEW

File: sanic/app.py [hunk 1]
```diff
@@ -737,15 +738,14 @@ async def handle_request(self, request: Request):
             request.route = route
 
             if (
-                request.stream.request_body  # type: ignore
+                request.stream
+                and request.stream.request_body
                 and not route.ctx.ignore_body
             ):
```
Comment (Tronic) [inline sanic/app.py:744]: This change seems unrelated, accidentally included in this PR?

Comment (ahopkins) [reply sanic/app.py:744]: `mypy` was complaining. We could use a type ignore, but I think this should be a light enough change.

**Review by Tronic — APPROVED** [review]
